test/functional/page_objects/visual_builder_page.ts

```diff
@@ -732,4 +821,37 @@ export class VisualBuilderPageObject extends FtrService {
     const optionInput = await this.testSubjects.find(`filterRatio${optionType}Input`);
     await optionInput.type(query);
   }
+
+  public async toggleNewChartsLibraryWithDebug(enabled: boolean) {
+    await this.kibanaServer.uiSettings.update({
+      'visualization:visualize:legacyChartsLibrary': !enabled,
+    });
+    await this.elasticChart.setNewChartUiDebugFlag(enabled);
+  }
+
+  public async getXAxisTitle() {
+    const [xAxis] = (await this.elasticChart.getChartDebugData())?.axes?.x ?? [];
+    return xAxis.title;
+  }
+
+  public async getLegendName(nth: number = 0) {
+    const legendItems = (await this.elasticChart.getChartDebugData())?.legend?.items ?? [];
+    return legendItems[nth]?.name;
+  }
+
+  public async getChartItemColor(nth: number = 0, itemType: 'areas' | 'bars' | 'lines' = 'areas') {
```

Comment:
nit: for me it looks a little bit extra especially in then you are using `getChartItemColor` and `getChartItemsCount` together. I think we can make out tests a little bit faster,  what do you think about: 

```
 public getChartItems(itemType: 'areas' | 'bars' | 'lines' = 'areas') {
    return this.elasticChart.getChartDebugData())?.[itemType] ?? [];
 }

const chartItems = getChartItems();
const itemsCount = chartItems.length;
const thirdItemColor = chartItems[3]?.color;
```

Reply:
can `itemType` be extended to work with `legend` too?